test/test_boost_is_constant_evaluated.cpp

```diff
@@ -0,0 +1,60 @@
+//  (C) Copyright John Maddock 2021.
```

Comment:
Would it be good to add a "sanity test"?

For some compiler you know it should have proper machinery in C++20 mode? I am not sure if boost tests have modern compilers running with -std=c++20, but I was thinking about something like this.

https://www.godbolt.org/z/r9Y5x3x6M

(use of assert just as an example, it can be a test framework assert_eq in real code)